provider/infoblox.go

```diff
@@ -0,0 +1,384 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	ibclient "github.com/infobloxopen/infoblox-go-client"
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	"github.com/kubernetes-incubator/external-dns/plan"
+)
+
+// IBConnector is an interface of ibclient.IBConnector that can be stubbed for testing.
+type IBConnector interface {
+	CreateObject(obj ibclient.IBObject) (ref string, err error)
+	GetObject(obj ibclient.IBObject, ref string, res interface{}) (err error)
+	DeleteObject(ref string) (refRes string, err error)
+	UpdateObject(obj ibclient.IBObject, ref string) (refRes string, err error)
+}
```

Comment:
@mikkeloscar, thanks for the review! This would be a case of monkey see, monkey do following my attempt to emulate the other providers. I understand now how this is superfluous.